packages/plugin/src/estree-parser/estree-ast.ts

```diff
@@ -1,31 +1,30 @@
 import { ASTNode, Location, TypeInfo, TypeNode, ValueNode } from 'graphql';
 import { BaseNode } from 'estree';
 
-export type SafeGraphQLType<T extends ASTNode | ValueNode> = Omit<
+type SafeGraphQLType<T extends ASTNode | ValueNode> = Omit<
   T extends { readonly type: TypeNode } ? Omit<T, 'type'> & { readonly gqlType: TypeNode } : T,
   'loc'
 >;
 
+export type TypeInformation = {
+  argument: ReturnType<TypeInfo['getArgument']>;
+  defaultValue: ReturnType<TypeInfo['getDefaultValue']>;
+  directive: ReturnType<TypeInfo['getDirective']>;
+  enumValue: ReturnType<TypeInfo['getEnumValue']>;
+  fieldDef: ReturnType<TypeInfo['getFieldDef']>;
+  inputType: ReturnType<TypeInfo['getInputType']>;
+  parentInputType: ReturnType<TypeInfo['getParentInputType']>;
+  parentType: ReturnType<TypeInfo['getParentType']>;
+  gqlType: ReturnType<TypeInfo['getType']>;
+};
+
 export type SingleESTreeNode<T, WithTypeInfo extends boolean> = T extends ASTNode | ValueNode
   ? SafeGraphQLType<T> &
       Pick<BaseNode, 'leadingComments' | 'loc' | 'range'> & {
         type: T['kind'];
-        gqlLocation: Location;
```

Comment:
this is typo, must be `Pick<Location, 'start' | 'end'>`

Reply:
Update: I removed this field because is unused in `graphql-eslint`, also this info can be retrieved with` range` field